subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/DirectoryWalkerTest.groovy

```diff
@@ -17,47 +17,26 @@
 package org.gradle.api.internal.file.collections
 
 import com.google.common.base.Charsets
-import org.gradle.api.GradleException
 import org.gradle.api.JavaVersion
-import org.gradle.api.file.FileTreeElement
 import org.gradle.api.file.FileVisitDetails
 import org.gradle.api.file.FileVisitor
-import org.gradle.api.internal.cache.StringInterner
 import org.gradle.api.internal.file.TestFiles
 import org.gradle.api.internal.file.collections.jdk7.Jdk7DirectoryWalker
-import org.gradle.api.specs.Spec
 import org.gradle.api.tasks.util.PatternSet
 import org.gradle.internal.Factory
-import org.gradle.internal.MutableBoolean
-import org.gradle.internal.snapshot.DirectorySnapshot
-import org.gradle.internal.snapshot.FileSystemLocationSnapshot
-import org.gradle.internal.snapshot.FileSystemSnapshotVisitor
-import org.gradle.internal.snapshot.impl.DirectorySnapshotter
 import org.gradle.test.fixtures.file.TestFile
-import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
 import org.gradle.util.Requires
-import org.gradle.util.SetSystemProperties
-import org.gradle.util.TestPrecondition
 import org.gradle.util.UsesNativeServices
-import org.junit.Rule
 import spock.lang.Issue
-import spock.lang.Specification
-import spock.lang.Unroll
 
 import java.nio.charset.Charset
 import java.util.concurrent.atomic.AtomicInteger
 
 @UsesNativeServices
-class DirectoryWalkerTest extends Specification {
-    @Rule
-    public final TestNameTestDirectoryProvider tmpDir = new TestNameTestDirectoryProvider()
-
-    @Rule
-    SetSystemProperties setSystemPropertiesRule
-
-
-    def cleanup() {
-        Charset.defaultCharset = null // clear cache
+class DirectoryWalkerTest extends AbstractDirectoryWalkerTest<DirectoryWalker> {
+    @Override
+    protected List<DirectoryWalker> getWalkers() {
+        return [new DefaultDirectoryWalker(), new Jdk7DirectoryWalker(), new ReproducibleDirectoryWalker()]
```

Comment:
The test `check that JDK7 walker gets picked with Unicode encoding as default` can be removed now, given that we don't support running on Java 7 any more.

Reply:
Let's do this in a separate PR.